lib/gis_robot_suite/descriptive_metadata_builder.rb

```diff
@@ -0,0 +1,595 @@
+      return unless node.any?
+
+      { value: node.first.text, type: 'other', displayLabel: 'Supplemental information' } # supplementalInformation is not repeatable in ISO19139
+    end
+
+    def subject
+      [].tap do |subjects|
+        subjects << keyword_subjects
+        subjects << category_subjects
+        subjects << temporal_subjects
+        subjects << coordinates_subjects
+      end.compact.flatten
+    end
+
+    def keyword_subjects
+      keywords = data_id_node.xpath('gmd:descriptiveKeywords/gmd:MD_Keywords/gmd:keyword/gco:CharacterString', NS)
+      return unless keywords.any?
+
+      keywords.map do |keyword|
+        type = keyword_type(keyword)
+        next unless type
+
+        { value: keyword.text,
+          type:,
+          source: keyword_source(keyword) }.compact
+      end.compact
+    end
+
+    def keyword_type(keyword)
+      code = keyword.xpath('../../gmd:type/gmd:MD_KeywordTypeCode', NS).text
+
+      if code == 'theme'
+        'topic'
+      elsif code == 'place'
+        'place'
+      end
+    end
+
+    def keyword_source(keyword)
+      return unless keyword.xpath('../../gmd:thesaurusName/gmd:CI_Citation', NS).any?
+
+      code = keyword.xpath('../../gmd:thesaurusName/gmd:CI_Citation/gmd:title/gco:CharacterString', NS).text
+      uri = keyword.xpath('../../gmd:thesaurusName/gmd:CI_Citation/gmd:identifier/gmd:MD_Identifier/gmd:code/gco:CharacterString', NS).text
+      return { code:, uri: } unless code.include?('Library of Congress Subject Headings (LCSH)') # preserving logic from xslt to handle older records
+
+      { code: 'lcsh', uri: }
+    end
+
+    def category_subjects
+      nodes = data_id_node.xpath('gmd:topicCategory/gmd:MD_TopicCategoryCode', NS)
+      return unless nodes.any?
+
+      nodes.map do |node|
+        label = CategoryMapper.label(node.text)
+        continue if label.blank? # do not include subject if topic not mapped to label
+
+        { source: { code: 'ISO19115TopicCategory', uri: 'http://www.isotc211.org/2005/resources/Codelist/gmxCodelists.xml#MD_TopicCategoryCode' },
+          value: label,
+          uri: node.text,
+          type: 'topic' }
+      end
+    end
+
+    def temporal_subjects
+      nodes = data_id_node.xpath('gmd:extent/gmd:EX_Extent/gmd:temporalElement/gmd:EX_TemporalExtent/gmd:extent', NS)
+      return unless nodes.any?
+
+      nodes.map do |node|
+        { value: extract_time(node),
+          type: 'time',
+          encoding: { code: 'w3cdtf' } }
+      end
+    end
+
+    def extract_time(node)
+      if node.xpath('gml:TimePeriod', NS).any?
+        begin_date = Date.parse(node.xpath('gml:TimePeriod/gml:beginPosition', NS).text).strftime('%Y')
+        end_date = Date.parse(node.xpath('gml:TimePeriod/gml:endPosition', NS).text).strftime('%Y')
+        "#{begin_date}-#{end_date}"
+      else
+        Date.parse(node.xpath('gml:TimeInstant/gml:timePosition', NS).text).strftime('%Y')
+      end
+    end
+
+    def coordinates_subjects
```

Comment:
Previously, "human-readable" coordinates in degrees were created in generate-mods, for both the native projection and re-projected to EPSG::4326. Now only create in the native projection.